--- /cvmfs/atlas.cern.ch/repo/sw/software/21.2/AnalysisTop/21.2.58/InstallArea/x86_64-slc6-gcc62-opt/src/Trigger/TrigAnalysis/TrigGlobalEfficiencyCorrection/Root/Calculator.cxx	2018-12-14 03:07:40.000000000 +0100
+++ Calculator.cxx	2019-02-12 11:59:43.000000001 +0100
@@ -106,11 +106,11 @@
 {
 	auto period = std::find_if(m_periods.cbegin(), m_periods.cend(),
 		[=](const Period& p) { return runNumber>=p.m_boundaries.first && runNumber<=p.m_boundaries.second; });
-	if(period == m_periods.end())
-	{
-		ATH_MSG_ERROR("No trigger combination has been specified for run " << runNumber);
-		return nullptr;
-	}
+	// if(period == m_periods.end())
+	// {
+	// 	ATH_MSG_ERROR("No trigger combination has been specified for run " << runNumber);
+	// 	return nullptr;
+	// }
 	return &*period;
 }
 
@@ -1403,4 +1403,4 @@
 	}
 
 	return false;
-}
\ ファイル末尾に改行がありません
+}
