--- Root/TopMuonCPTools.cxx.orig	2018-05-29 20:07:08.000000001 +0200
+++ Root/TopMuonCPTools.cxx	2018-05-29 20:08:52.000000001 +0200
@@ -327,9 +327,25 @@
     tool = asg::ToolStore::get<CP::MuonEfficiencyScaleFactors>(name);
   } else {
     tool = new CP::MuonEfficiencyScaleFactors(name);
-    top::check(asg::setProperty(tool, "WorkingPoint", WP),
+    
+	if(WP == "PromptLeptonIso" || WP == "PromptLepton"){
+        std::cout << "ALBERT WP: " << WP << std::endl;
+
+        top::check(asg::setProperty(tool, "WorkingPoint",WP),
                 "Failed to set WP for " + name + " tool");
-    top::check(tool->initialize(),
+        char* const rc = getenv("WorkDir_DIR");
+
+        top::check(asg::setProperty(tool, "CustomInputFolder",
+        	std::string(rc)+"/data/ttHMultilepton/data/R21_PLV_SFs"),
+        	"Failed to Set Custom folder for " + name + " tool");
+
+        top::check(asg::setProperty(tool, "CustomFileCombined", "Iso_mptLeptonVetoIso_Z.root"),
+        	"Failed to set Custom file for " + name + " tool");
+    } else {
+    	top::check(asg::setProperty(tool, "WorkingPoint", WP),
+    		"Failed to set WP for " + name + " tool");
+    }
+	top::check(tool->initialize(),
                 "Failed to set initialize " + name);
   }
   return tool;
