Index: Root/TopEgammaCPTools.cxx
===================================================================
--- Root/TopEgammaCPTools.cxx	(Revision 806266)
+++ Root/TopEgammaCPTools.cxx	(Arbeitskopie)
@@ -419,10 +419,15 @@
   } else if (type == "isolation") {
     file_path = "isolation/efficiencySF.Isolation."
       + ID + "_d0z0_v11" + new_iso + ".root";
+    if (isolation == "PromptLepton" || isolation == "PromptLeptonCFT") { // should be only true in case of using ttH-ML private root file
+      return PathResolverFindCalibFile("ttHMultilepton/efficiencySF.Isolation.TightLLH_d0z0_v11_isol" + isolation + ".root");
+    }
   } else if (type == "triggerSF") {
+    if (isolation == "PromptLepton" || isolation == "PromptLeptonCFT") new_iso = "";
     file_path = "trigger/efficiencySF." + ridiculous_trigger_string
       + "." + ID + "_d0z0_v11" + new_iso + ".root";
   } else if (type == "triggerEff") {
+    if (isolation == "PromptLepton" || isolation == "PromptLeptonCFT") new_iso = "";
     file_path = "trigger/efficiency." + ridiculous_trigger_string
       + "." + ID + "_d0z0_v11" + new_iso + ".root";
   } else if (type == "ChargeID") {
