--- ../../../../hackJES/TopJetMETCPTools_original36.cxx	2019-04-04 14:04:11.554379450 +0200
+++ ../../../TopCPTools/Root/TopJetMETCPTools.cxx	2019-04-05 16:46:50.432257000 +0200
@@ -134,11 +134,18 @@
     ATH_MSG_INFO("Insitu corrections for data are not yet available and not neglible");
 
     // Data
-    m_jetAntiKt4_Data_ConfigFile          = "JES_data2017_2016_2015_Recommendation_Feb2018_rel21.config";
+    //m_jetAntiKt4_Data_ConfigFile          = "JES_data2017_2016_2015_Recommendation_Feb2018_rel21.config";
+    //new Nov18 release: 
+    m_jetAntiKt4_Data_ConfigFile          = "JES_data2017_2016_2015_Consolidated_EMTopo_2018_Rel21.config";
     m_jetAntiKt4_Data_CalibSequence       = "JetArea_Residual_EtaJES_GSC_Insitu";
+
     // FS EM/LC
-    m_jetAntiKt4_MCFS_ConfigFile          = "JES_data2017_2016_2015_Recommendation_Feb2018_rel21.config";
-    m_jetAntiKt4_MCFS_CalibSequence       = "JetArea_Residual_EtaJES_GSC";
+    //m_jetAntiKt4_MCFS_ConfigFile          = "JES_data2017_2016_2015_Recommendation_Feb2018_rel21.config";
+    //m_jetAntiKt4_MCFS_CalibSequence       = "JetArea_Residual_EtaJES_GSC";
+    //new Nov18 release: 
+    m_jetAntiKt4_MCFS_ConfigFile          = "JES_data2017_2016_2015_Consolidated_EMTopo_2018_Rel21.config";
+    m_jetAntiKt4_MCFS_CalibSequence       = "JetArea_Residual_EtaJES_GSC_Smear";
+
     // AFII EM/LC
     m_jetAntiKt4_MCAFII_ConfigFile        = "JES_MC16Recommendation_AFII_EMTopo_April2018_rel21.config";
     m_jetAntiKt4_MCAFII_CalibSequence     = "JetArea_Residual_EtaJES_GSC";
@@ -179,6 +186,9 @@
     m_jetCalibrationTool = asg::ToolStore::get<IJetCalibrationTool>("JetCalibrationTool");
   } else {
     std::string calibConfig, calibSequence;
+    //
+    std::string calibArea="00-04-82";
+
     if (m_config->isMC()) {
       // AFII
       if (m_config->isAFII()) {
@@ -218,6 +228,10 @@
                 "Failed to set ConfigFile " + calibConfig);
     top::check(asg::setProperty(jetCalibrationTool, "CalibSequence", calibSequence),
                 "Failed to set CalibSequence " + calibSequence);
+
+    top::check(asg::setProperty(jetCalibrationTool, "CalibArea", calibArea),
+               "Failed to set CalibArea " + calibArea);
+
     top::check(asg::setProperty(jetCalibrationTool, "IsData", !m_config->isMC()),
                 "Failed to set IsData " + !m_config->isMC());
 
