--- /cvmfs/atlas.cern.ch/repo/sw/software/21.2/AnalysisTop/21.2.58/InstallArea/x86_64-slc6-gcc62-opt/src/PhysicsAnalysis/TopPhys/xAOD/TopCPTools/Root/TopMuonCPTools.cxx	2018-12-14 03:07:38.000000000 +0100
+++ TopMuonCPTools.cxx	2019-01-10 10:24:28.000000001 +0100
@@ -285,9 +285,25 @@
     tool = asg::ToolStore::get<CP::MuonEfficiencyScaleFactors>(name);
   } else {
     tool = new CP::MuonEfficiencyScaleFactors(name);
-    top::check(asg::setProperty(tool, "WorkingPoint", WP),
+    
+	if(WP == "PromptLeptonIso" || WP == "PromptLepton"){
+        std::cout << "ALBERT WP: " << WP << std::endl;
+
+        top::check(asg::setProperty(tool, "WorkingPoint",WP),
                 "Failed to set WP for " + name + " tool");
-    top::check(tool->initialize(),
+        char* const rc = getenv("WorkDir_DIR");
+
+        top::check(asg::setProperty(tool, "CustomInputFolder",
+        	std::string(rc)+"/data/ttHMultilepton/data/R21_PLV_SFs"),
+        	"Failed to Set Custom folder for " + name + " tool");
+
+        top::check(asg::setProperty(tool, "CustomFileCombined", "Iso_mptLeptonVetoIso_Z.root"),
+        	"Failed to set Custom file for " + name + " tool");
+    } else {
+    	top::check(asg::setProperty(tool, "WorkingPoint", WP),
+    		"Failed to set WP for " + name + " tool");
+    }
+	top::check(tool->initialize(),
                 "Failed to set initialize " + name);
   }
   return tool;
