--- ../../../TopCPTools/Root/TopFlavorTaggingCPTools.cxx	2018-07-02 09:44:22.000000001 +0200
+++ asd.cxx	2018-07-02 09:56:17.000000001 +0200
@@ -42,7 +42,7 @@
   }
 
   m_tagger          = ""; // Extract in the loop
-  m_cdi_file        = "xAODBTaggingEfficiency/13TeV/2017-21-13TeV-MC16-CDI-2018-05-04_v1.root";
+  m_cdi_file        = "xAODBTaggingEfficiency/13TeV/2017-21-13TeV-MC16-CDI-2018-06-24_v1.root";
   // This ordering needs to match the indexing in TDP (for missing cases, we use default which gives a MC/MC of 1 as its the same as the eff used in the calibration
   // Pythia6;Herwigpp;Pythia8;Sherpa(2.2);Sherpa(2.1)
   m_efficiency_maps = "default;410558;410501;410250;default";
